### ISSUE TITLE
Force Max Resolution Thumbnail

### ISSUE THREAD

**Issue by ParticleCore** [ParticleCore/Iridium#742]
Implement feature that forces the use of max resolution thumbnail to show on the video player when the auto play is turned off.

**Comment by ParticleCore** [comment]
When the video playing mode is set to `cue` (autoplay disbled) the following code is responsible for loading the video thumbnail with a respective size depending on the size of the player

````javascript
    g.h.HL = function(a) {
        for (var b = [], c = 0; c < arguments.length; ++c)
            b[c - 0] = arguments[c];
        if (!this.l.getVideoData().Kv) {
            b = g.p(b);
            for (c = b.next(); !c.done; c = b.next())
                if (c = c.value,
                !(!c || c instanceof g.EB && !c.videoId)) {
                    var d = g.dK(this.l).getPlayerSize();
                    var e = g.Ut()
                      , f = d.width * e;
                    e *= d.height;
                    if (1280 < f || 720 < e)
                        if (d = c.dc("maxresdefault.jpg"))
                            break;
                    if (640 < f || 480 < e)
                        if (d = c.dc("maxresdefault.jpg"))
                            break;
                    if (320 < f || 180 < e)
                        if (d = c.dc("sddefault.jpg") || c.dc("hqdefault.jpg") || c.dc("mqdefault.jpg"))
                            break;
                    if (d = c.dc("default.jpg"))
                        break
                }
            this.C.style.backgroundImage = d ? "url(" + d + ")" : ""
        }
    };
````

In order to load the highest quality version a sub-routine is implemented right before the URL is set for the background image in order to check if the `maxresdefault` thumbnail exists and if it is exists it will update the background image with the new URL.

Example of how this can be done as follows:

````javascript
    g.h.HL = function(a) {
        for (var b = [], c = 0; c < arguments.length; ++c)
            b[c - 0] = arguments[c];
        if (!this.l.getVideoData().Kv) {
            b = g.p(b);
            for (c = b.next(); !c.done; c = b.next())
                if (c = c.value,
                !(!c || c instanceof g.EB && !c.videoId)) {
                    var d = g.dK(this.l).getPlayerSize();
                    var e = g.Ut()
                      , f = d.width * e;
                    e *= d.height;
                    if (1280 < f || 720 < e)
                        if (d = c.dc("maxresdefault.jpg"))
                            break;
                    if (640 < f || 480 < e)
                        if (d = c.dc("maxresdefault.jpg"))
                            break;
                    if (320 < f || 180 < e)
                        if (d = c.dc("sddefault.jpg") || c.dc("hqdefault.jpg") || c.dc("mqdefault.jpg"))
                            break;
                    if (d = c.dc("default.jpg"))
                        break
                }
            this.C.style.backgroundImage = d ? "url(" + d + ")" : "";
            (function imageLoader(element, url) {

                if (!url || !url.match(/default\.[a-z]+/)) {
                    return;
                }

                function checkHighQualityThumbnail(event) {
                    if (event.target.width > 120) {

                        element.parentElement.style.backgroundImage = `url(${url})`;
                        element.style.backgroundImage = `url(${src})`;

                    }
                }

                let src;
                let thumbnail;

                src = url.replace(/[a-z]+default(\.[a-z]+)/g, "maxresdefault$1");
                thumbnail = new Image();
                thumbnail.addEventListener("load", checkHighQualityThumbnail, false);
                thumbnail.src = src;

            }(this.C, d));
        }
    };
````

 - The string replacement in the URL is only done on the filename without touching the extension since it can be `.jpg` as well as `.webp` or something else that is not unaccounted for
 - The fallback thumbnail size is 120x90 ([example](https://i.ytimg.com/vi/_JgyR-N_tt5/maxresdefault.jpg)), as such if the image returned is not bigger than 120px in width the video does not have a higher quality thumbnail available and the default one is left untouched
- In order to ensure a smooth transition we apply the normal background image to the parent element of the target element. The reason for this is once the highest resolution thumbnail URL is set the browser still has to load the image thus resulting in a black flicker (or longer if the connection is slow) when switching between the low and high resolution images. Since the parent element has the original low resolution image, once the highest resolution image is set on the target element it will still display the original low resolution image behind it until the highest one finishes loading resulting in an almost seamless transition with minimal setup

**Comment by ParticleCore** [comment]
@BobbyWibowo Apologies for pinging you, but since you've been using this could you tell me if it is better? It should no longer be showing the 3 dots default image in case the high resolution isn't available.

**Comment by BobbyWibowo** [comment]
@ParticleCore Ah yes, I actually noticed and started using https://github.com/ParticleCore/Iridium/commit/3b1a5765475948c4cb70d6e168eca44188ece16a since 4~5 days ago.
I can confirm that it's showing the default thumbs on videos with no max res thumbnails, and will load them on the ones who do 👍
Of course there's the typical internet half a sec delay or so before the thumbs turn max res on those videos, but no problem with those at all.

**Comment by ParticleCore** [comment]
@BobbyWibowo Glad to hear you like the settings page. The version I just pushed should be the final UI version, I don't plan to change much after that. All that's left is to finish locale implementation and the remaining features.

If you have any suggestions for the settings page feel free to add them here: #739
I'd be more than happy to review them and see if there is anything worth adding to it.

**Comment by ParticleCore** [comment]
Implemented in 1.0.0
